src/experiments.js

```diff
@@ -226,8 +222,63 @@ Test.registerGoal( 'click', ( element, record ) => {
 	} );
 }, [ 'a' ] );
 
+class ExperienceBlock extends HTMLElement {
+
+	get clientId() {
+		return this.getAttribute( 'client-id' );
+	}
+
+	connectedCallback() {
+		// Set default styles.
+		this.attachShadow( { mode: 'open' } );
+		this.shadowRoot.innerHTML = `
+			<style>
+				:host {
+					display: block;
+				}
+			</style>
+			<slot></slot>
+		`;
+
+		// Update the component content.
+		this.setContent();
+
+		// Attach a listener to update the content when audiences are changed.
+		window.Altis.Analytics.on( 'updateAudiences', this.setContent );
+	}
+
+	setContent() {
+		const audiences = window.Altis.Analytics.getAudiences();
+
+		// Get associated templates.
+		const templates = document.querySelectorAll( `template[data-parent-id="${ this.clientId }"]` );
+
+		// Find a matching template.
+		for ( let index = 0; index < templates.length; index++ ) {
+			const template = templates[ index ];
+			const audience = Number( template.dataset.audience );
+			if ( audiences.indexOf( audience ) >= 0 ) {
+				const experience = template.content.cloneNode( true );
+				this.innerHTML = '';
+				this.appendChild( experience );
+
+				// Log an event for tracking views and audience.
+				window.Altis.Analytics.record( 'experienceView', {
+					attributes: {
+						clientId: this.clientId,
```

Comment:
Should be also push the variant?

Reply:
Yep that's what the `audience` is on the line below.

Reply:
Ahh I had assumed this was the audience that was matched, not the variantId, is it worth renaming in that case? Or am I missing something? I was thinking a variant has an ID, and can be for multiple audiences

Reply:
Right now it's one audience per variant, and that value could be an array anyway. The variant ID itself isn't really that important, rather how it's selected is. We could store the variant client IDs as well and track them easily enough but audience is the key differentiator here.